liota/lib/transports/web_socket.py

```diff
@@ -57,84 +57,81 @@ def connect_soc(self):
+                except ssl.SSLError:
+                    pass
+            if self.ws is None:
+                raise (IOError("Couldn't verify host certificate"))
 
     def run(self):
         try:
             log.debug("Stream Opened")
             while True:
                 msg = self.ws.recv()
-                log.debug ("Message received while running {0}".format(msg))
+                log.debug("Message received while running {0}".format(msg))
                 if msg is "":
-                    # TO DO Check if os._exit(0) required everywhere
                     log.error("Stream Closed")
-                    os._exit(0)
+                    sys.exit(0)
                     break
                 log.debug("RX {0}".format(msg))
                 if self.on_receive is not None:
                     self.on_receive(msg)
         except Exception:
             log.exception("Exception on receiving the response from Server, please check the connection and try again.")
             self.close()
-            os._exit(0)
+            sys.exit(0)
 
     def send(self, msg):
-      request_calls = ['request', 'response']
-      complete_message = json.dumps(msg)
-      log.info("TX Sending message {0}".format(complete_message))
-      try:
-          self.ws.send(complete_message)
-      except:
-          # Retry logic only for publishing stats, not for request or response calls
-          if all(request not in complete_message for request in request_calls):
-              attempts = 1
-              while attempts < 4:
-                  try:
-                      log.info("Exception while sending data, applying retry logic.")
-                      self.connect_soc()
-                      log.info("Created New Websocket")
-                      log.info("TX Sending message {0}".format(complete_message))
-                      self.ws.send(complete_message)
-                      break
-                  except Exception:
-                      # Three times retry websocket connection for publishing data
-                      log.info("{0} attempt".format(attempts))
-                      attempts += 1
-                      if attempts == 4:
-                          log.exception("Exception while sending data, please check the connection and try again.")
-                          self.close()
-                          os._exit(0)
-          else:
-              log.exception("Exception while sending data, please check the connection and try again.")
-              self.close()
-              os._exit(0)
+        request_calls = ['request', 'response']
+        complete_message = json.dumps(msg)
+        log.info("TX Sending message {0}".format(complete_message))
+        try:
+            self.ws.send(complete_message)
+        except:
+            # Retry logic only for publishing stats, not for request or response calls
+            if all(request not in complete_message for request in request_calls):
```

Comment:
Can be done but over here we have specific self.ws.send used to publish the message after the websocket object is re-created (can be revisited later).